compiler/rustc_trait_selection/src/traits/query/evaluate_obligation.rs

```diff
@@ -46,6 +49,12 @@ impl<'cx, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'cx, 'tcx> {
         &self,
         obligation: &PredicateObligation<'tcx>,
     ) -> bool {
+        if let PredicateKind::Trait(pred) = obligation.predicate.kind().skip_binder() {
+            if let hir::Constness::Const = pred.constness {
+                // do not evaluate to holds when we have a const predicate.
```

Comment:
What is going on here? Why aren't we checking const trait predicates?

Reply:
Well... This is an awkward fix. The problem is without this check to prevent us from using the cache and so on, the ui tests would fail. I might have time tomorrow to try to find a better way to approach this.

Reply:
Seems fine for now, but document the behaviour + whatever you remember about details.